### PR TITLE
Add support for `SENTRY_RELEASE` in bash-hook

### PR DESCRIPTION
**Pull request by boozec**
This PR adds support for `SENTRY_RELEASE` variable in the `bash-hook` command.
It adds `--release` argument in the command but the environment variable is
still available, so it checks the value from env first.

Resolves https://github.com/getsentry/sentry-cli/issues/1674

### CODE REVIEW

**Comment by boozec** [comment]
Clippy lint fails for some old sources, should I fix them anyway?

PS: on local instance it does not fail :\

**Comment by kamilogorek** [comment]
It fails because 4 days ago, Rust `1.72.0` was released, and as always, Clippy came with new defaults 😄

**Comment by boozec** [comment]
Ops, I missed the update to `1.72.0`, thank you :)

File: src/commands/bash_hook.rs [hunk 1]
```diff
@@ -68,15 +68,30 @@ pub fn make_command(command: Command) -> Command {
...
+    environ: bool,
+) -> Result<()> {
     let config = Config::current();
 
     let mut event = Event {
         environment: config.get_environment().map(Into::into),
-        release: detect_release_name().ok().map(Into::into),
+        release: release.map_or(detect_release_name().ok().map(Into::into), |release| {
+            Some(release).map(Into::into)
+        }),
```
Comment (kamilogorek) [inline src/commands/bash_hook.rs:94]:
```suggestion
        release: release.or(detect_release_name().ok()).map(Into::into),
```

File: src/commands/bash_hook.rs [hunk 2]
```diff
@@ -192,6 +207,13 @@ fn send_event(traceback: &str, logfile: &str, tags: &[&String], environ: bool) -
 }
 
 pub fn execute(matches: &ArgMatches) -> Result<()> {
+    let config = Config::current();
+    let release = if let Ok(release) = config.get_release(matches) {
+        Some(release)
+    } else {
+        None
+    };
```
Comment (kamilogorek) [inline src/commands/bash_hook.rs:215]:
```suggestion
    let release = Config::current().get_release(matches).ok();
```

File: src/commands/bash_hook.rs [hunk 3]
```diff
@@ -228,6 +251,14 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
             .join(""),
     );
 
+    script = match release {
+        Some(release) => script.replace(
+            "___SENTRY_RELEASE___",
+            format!(" --release \"{}\"", release).as_str(),
+        ),
+        None => script.replace("___SENTRY_RELEASE___", ""),
```
Comment (kamilogorek) [inline src/commands/bash_hook.rs:259]:
```suggestion
            " ___SENTRY_RELEASE___",
            format!(" --release \"{}\"", release).as_str(),
        ),
        None => script.replace(" ___SENTRY_RELEASE___", ""),
```

File: src/bashsupport.sh [hunk 4]
```diff
@@ -32,7 +32,7 @@ _sentry_err_trap() {
   echo "@exit_code:${_exit_code}" >> "$_SENTRY_TRACEBACK_FILE"
 
   : >> "$_SENTRY_LOG_FILE"
-  export SENTRY_LAST_EVENT=$(___SENTRY_CLI___ bash-hook --send-event --traceback "$_SENTRY_TRACEBACK_FILE" ___SENTRY_TAGS___ --log "$_SENTRY_LOG_FILE" ___SENTRY_NO_ENVIRON___)
+  export SENTRY_LAST_EVENT=$(___SENTRY_CLI___ bash-hook --send-event --traceback "$_SENTRY_TRACEBACK_FILE" ___SENTRY_TAGS______SENTRY_RELEASE___ --log "$_SENTRY_LOG_FILE" ___SENTRY_NO_ENVIRON___)
```
Comment (kamilogorek) [inline src/bashsupport.sh:35]:
```suggestion
  export SENTRY_LAST_EVENT=$(___SENTRY_CLI___ bash-hook --send-event --traceback "$_SENTRY_TRACEBACK_FILE" ___SENTRY_TAGS___ ___SENTRY_RELEASE___ --log "$_SENTRY_LOG_FILE" ___SENTRY_NO_ENVIRON___)
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: src/commands/bash_hook.rs [hunk 5]
```diff
@@ -68,15 +68,30 @@ pub fn make_command(command: Command) -> Command {
                 .action(ArgAction::Append)
                 .help("Add tags (key:value) to the event."),
         )
+        .arg(
+            Arg::new("release")
+                .value_name("RELEASE_VERSION")
```
Comment (kamilogorek) [inline src/commands/bash_hook.rs:73]:
```suggestion
                .value_name("RELEASE")
```

File: tests/integration/_cases/bash_hook/bash_hook-help.trycmd [hunk 6]
```diff
@@ -6,18 +6,19 @@ Prints out a bash script that does error handling.
...
+      --no-environ                 Do not send environment variables along
+      --auth-token <AUTH_TOKEN>    Use the given Sentry auth token.
+      --cli <CMD>                  Explicitly set/override the sentry-cli command
+      --log-level <LOG_LEVEL>      Set the log output verbosity. [possible values: trace, debug,
+                                   info, warn, error]
+      --quiet                      Do not print any output while preserving correct exit code. This
+                                   flag is currently implemented only for selected subcommands.
+                                   [aliases: silent]
+      --tag <KEY:VALUE>            Add tags (key:value) to the event.
+      --release <RELEASE_VERSION>  Define release version for the event.
```
Comment (kamilogorek) [inline tests/integration/_cases/bash_hook/bash_hook-help.trycmd:21]:
```suggestion
      --release <RELEASE>  Define release version for the event.
```

**Review by kamilogorek — CHANGES_REQUESTED** [review]

File: src/commands/bash_hook.rs (hunk 1 above)
Comment (kamilogorek) [reply src/commands/bash_hook.rs:94]: Simplified

File: src/commands/bash_hook.rs (hunk 2 above)
Comment (kamilogorek) [reply src/commands/bash_hook.rs:215]: Simplified

File: src/commands/bash_hook.rs (hunk 3 above)
Comment (kamilogorek) [reply src/commands/bash_hook.rs:259]: Better redability for `bashsupport.sh`

File: src/commands/bash_hook.rs (hunk 5 above)
Comment (kamilogorek) [reply src/commands/bash_hook.rs:73]: Consistency with other `--release` flags.

File: tests/integration/_cases/bash_hook/bash_hook-help.trycmd (hunk 6 above)
Comment (kamilogorek) [reply tests/integration/_cases/bash_hook/bash_hook-help.trycmd:21]: Consistency with other `--release` flags.

**Comment by boozec** [comment]
Thank you for all the simplification tips :)

**Pull request merged**